### PR TITLE
Creep integration

### PR DESCRIPTION
**Pull request by reverendbedford**
This adds a line search capability to the entire NEML nonlinear solver system.  It, however, only turns it on by default for the creep + plasticity models, in an effort to improve the robustness of the model integration.

### CODE REVIEW

**Review by arovinelli — COMMENTED** [review]
In general looks good, but I have a few general comments.

First I don't like much the fact that we don't have inputs declared as constants in all the methods, though this something widespread in NEML... we should be more carefull
Second, can we add a test were checking that we effectively stop the solver when the relative tolerance is achieved? 
Third, but this is more of a technicality? why don't we set the default tolerance to the same value for all the models? How do you know the tolerances you selected are the best ones (experience is valid answer)?

**Comment by reverendbedford** [comment]
1. I think I had trouble with NOX when I did a const-correct version of the solver.  We may want to take NOX out at some point, at which time I'll enter an IOU for fixing this issue.
2. Good idea.  See new commit.
3. Yes, basically experience.  But no, they can't be the same.  For example, the creep-plasticity model (which is a nightmare) requires a very different tolerance than the other models...

File: src/solvers.cxx [hunk 1]
```diff
@@ -13,24 +13,24 @@ namespace neml {
...
 #endif
 }
 
-int newton(Solvable * system, double * x, TrialState * ts,
-          double tol, int miter, bool verbose, bool relative,
-          double * R, double * J)
+int newton(Solvable * system, double * x, TrialState * ts, SolverParameters p, double * R,
+           double * J)
 {
+  int mline = 10;
```
Comment (arovinelli) [inline src/solvers.cxx:32]: didn't notice this before, but shouldn't this bee a controllable parameter?

Comment (reverendbedford) [reply src/solvers.cxx:32]: No, not right now.  We're doing really naive backtracking so I don't want them trying to mess with it.  When/if we add proper linesearch we'll want to make it configurable (along with the alpha value or whatever they usually use).

Comment (arovinelli) [reply src/solvers.cxx:32]: fair enough,
you might want to add a TODO comment then, but up to you.

If test passes I'll approve now

**Comment by reverendbedford** [comment]
@arovinelli tests pass

**Review by arovinelli — APPROVED** [review]
Looks good to me, approved

File: src/solvers.cxx (hunk 1 above)
Comment (reverendbedford) [reply src/solvers.cxx:32]: Thanks.

**Pull request merged**